File: 30days/day06-review-loop/Solution.cs
// PROBLEM: https://www.hackerrank.com/challenges/30-review-loop/problem

using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int lines = Convert.ToInt32(Console.ReadLine());
        for (int i = 1; i <= lines; i++) {
            string even = "";
            string odd = "";
            string line = Console.ReadLine();
            for (int ii = 0; ii < line.Length; ii++) {
                if (ii % 2 == 0) {
                    even += line[ii];
                }
                else {
                    odd += line[ii];
                }
            }
            Console.WriteLine($"{even} {odd}");
        }
    }
}

File: 30days/day06-review-loop/Solution.js
// PROBLEM: https://www.hackerrank.com/challenges/30-review-loop/problem

function processData(input) {
    input = input.split('\n');
    for (let i = 1; i < input.length; i++){
        let even = '';
        let odd = '';

        for(let ii = 0; ii < input[i].length; ii++){
            if(ii % 2 === 0){
                even += input[i][ii];
            }else{
                odd += input[i][ii];
            }
        }
        console.log(`${even} ${odd}`);
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
